Add contributing guidelines (#119)
ROCm/rocprofiler-systems: CONTRIBUTING.md

* If your issue is already listed, upvote the issue and
  comment or post to provide additional details, such as how you reproduced this issue.
* If you're not sure if your issue is the same, err on the side of caution and file your issue.
  You can add a comment to include the issue number (and link) for the similar issue. If we evaluate
  your issue as being the same as the existing issue, we'll close the duplicate.
* If your issue doesn't exist, use the issue template to file a new issue.
  * When filing an issue, be sure to provide as much information as possible, including script output so
    we can collect information about your configuration. This helps reduce the time required to
    reproduce your issue.
  * Check your issue regularly, as we may require additional information to successfully reproduce the
    issue.
* You may also open an issue to ask questions to the maintainers about whether a proposed change
  meets the acceptance criteria, or to discuss an idea pertaining to the library.

## Acceptance Criteria ##

* Contributions should align with the project's goals and maintainability.
* Code should be well-documented and include tests where applicable.

> * Ensure that your changes do not break existing functionality.

Reviewer: ```suggestion
* Ensure that your changes do not break existing functionality.
* Each commit is to be digitally signed. For more details see: [About commit signature verification - GitHub Docs](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification).
```
We should also mention that we require commits to be "verified" somewhere. Maybe here?

<head>
  <meta charset="UTF-8">
  <meta name="description" content="Contributing to rocprofiler-systemsE">

>   <meta name="keywords" content="ROCm, contributing, rocprofiler-systemsE">

Reviewer: ```suggestion
  <meta name="description" content="Contributing to rocprofiler-systems">
  <meta name="keywords" content="ROCm, contributing, rocprofiler-systems">
```
I wasn't sure. Was the "E" at the end required.

</head>

# Contributing to rocprofiler-systems #

ROCm Systems Profiler (rocprofiler-systems), formerly Omnitrace, is a comprehensive profiling and tracing tool for parallel applications written in C, C++, Fortran, HIP, OpenCL, and Python which execute on the CPU or CPU+GPU.

We welcome contributions to rocprofiler-systems.  Please follow these details to help ensure your contributions will be successfully accepted.

## Table of Contents ##

1. [Issue Discussion](#issue-discussion)
2. [Pull Request Guidelines](#pull-request-guidelines)
3. [Setting Up the Development Environment](#setting-up-the-development-environment)
4. [Running Tests](#running-tests)

> 5. [Code Structure](#code-structure)

Reviewer: ```suggestion
```
This section doesn't exist

6. [Coding Style](#coding-style)
7. [Code License](#code-license)

> 8. [Code of Conduct](#code-of-conduct)

Reviewer: ```suggestion
```
This section does not exist

  comment or post to provide additional details, such as how you reproduced this issue.
* If you're not sure if your issue is the same, err on the side of caution and file your issue.
  You can add a comment to include the issue number (and link) for the similar issue. If we evaluate
  your issue as being the same as the existing issue, we'll close the duplicate.
* If your issue doesn't exist, use the issue template to file a new issue.
  * When filing an issue, be sure to provide as much information as possible, including script output so
    we can collect information about your configuration. This helps reduce the time required to
    reproduce your issue.
  * Check your issue regularly, as we may require additional information to successfully reproduce the
    issue.
* You may also open an issue to ask questions to the maintainers about whether a proposed change
  meets the acceptance criteria, or to discuss an idea pertaining to the library.

## Acceptance Criteria ##

* Contributions should align with the project's goals and maintainability.
* Code should be well-documented and include tests where applicable.
* Ensure that your changes do not break existing functionality.

> ### Exceptions ###
> * If you believe your contribution does not fit the guidelines but is still valuable, please discuss it with the maintainers before submitting.

Reviewer: ```suggestion
### Exceptions ###

* If you believe your contribution does not fit the guidelines but is still valuable, please discuss it with the maintainers before submitting.
```
Small linting issues. There should be a newline between header and body. (see: https://github.com/DavidAnson/markdownlint/blob/v0.37.4/doc/md022.md)

* If you've added code that should be tested, add tests.
* Ensure the test suite passes.
* Make sure your code conforms to the format. Use clang-format-11 and/or cmake-format.
* Use clear and descriptive commit messages.
* Submit your PR and work with the reviewer or maintainer to get your PR approved
* Once approved, the PR is brought onto internal CI systems and may be merged into the component during our release cycle, as coordinated by the maintainer.

### Setting Up the Development Environment ###

* It is recommended to [fork](https://github.com/ROCm/rocprofiler-systems/fork) the repository.
* Clone your forked repository: `git clone https://github.com/ROCm/<yourgithub-id>/rocprofiler-systems.git`
* Navigate to the project directory: `cd rocprofiler-systems`
* Set the original repository URL as the remote upstream using `git remote add upstream https://github.com/ROCm/rocprofiler-systems` (or `git remote set-url upstream https://github.com/ROCm/rocprofiler-systems`)
* Verify if origin and upstream points correctly with `git remote -v`.
* Start a new branch for your work: `git checkout -b topic-<yourFeatureName>`

> * Build the project as outlined [here](https://github.com/ROCm/rocprofiler-systems/blob/a03770c0606c23fda5e2c83782f2d188eb8522f5/docs/install/install.rst#building-and-installing-rocm-systems-profiler).

Reviewer: ```suggestion
* Build the project as outlined in [ROCm documentation](docs/install/install.rst#building-and-installing-rocm-systems-profiler).
```
Or, something to avoid the "click here" formatting.

### Running Tests ###

* To run the test suite, use the following command: `make test`
* Ensure all tests pass before submitting a pull request.
* If the project was built with option `-D ROCPROFSYS_BUILD_TESTING=ON`, then the tests are built with it. Individual tests groups can be run using command: `ctest -R <test-name> -V --output-on-failure`. Command `ctest --print-labels` will list all the test names which can be passed to -R as test-name. 

## Coding Style ##

* Adhere to the coding style used in the project. This includes naming conventions, indentation, and commenting practices.
* Follow the existing directory structure and organization of the codebase.
* Group related files together and maintain a logical hierarchy.
* Use clang-format-11 and cmake-format formatters to ensure consistency.

## Code License ##

All code contributed to this project will be licensed under the license identified in the [License](LICENSE). Your contribution will be accepted under the same license.

## References ##

> 1. [ROCm Systems Profiler Wiki](https://github.com/ROCm/MIOpen/wiki)

Reviewer: ```suggestion
1. [ROCm Systems Profiler Documentation](https://rocm.docs.amd.com/projects/rocprofiler-systems/en/latest/index.html)
```

> 2. [ROCm Systems Profiler Readme](https://github.com/ROCm/rocprofiler-systems/blob/amd-mainline/README.md)

Reviewer: ```suggestion
2. [ROCm Systems Profiler README](README.md)
```

> 

Reviewer: ```suggestion
```
Small linting error. Multiple consecutive blank lines (See: https://github.com/DavidAnson/markdownlint/blob/v0.37.4/doc/md012.md)
